pkg/retrieval/retrieval.go

```diff
@@ -212,6 +232,12 @@ func (s *Service) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream) (e
 		return fmt.Errorf("write delivery: %w peer %s", err, p.Address.String())
 	}
 
+	chunkPrice := s.pricer.Price(chunk.Address())
+	err = s.accounting.Debit(p.Address, chunkPrice)
+	if err != nil {
+		return err
+	}
```

Comment:
No. Reserving is to prevent overspending. Here we are getting "paid" (by putting the peer in debt). It's the peer that reserves when sending the request.